added a UUID generator

FlorianZipser opened the pull request:
I added the UuidGenerator class to randomly generate UUIDs and a test for it. Further I registered the generator at the bean-matcher class but unfortunatlely reformatted the class :(.  It would be great if you could consider to merge the pull request.

orien commented on src/test/java/com/google/code/beanmatchers/UuidGeneratorTest.java:
```diff
@@ -0,0 +1,41 @@
+	}
+
+	@Test
+	public void shouldProvideString() {
```
Rename to `shouldProvideUuid`.

orien replied:
Rename to `shouldProvideUuid`.

orien commented on src/test/java/com/google/code/beanmatchers/UuidGeneratorTest.java:
```diff
@@ -0,0 +1,41 @@
+	}
+
+	@Test
+	public void shouldProvideDifferingStringOnSubsequentCall() {
```
Rename to `shouldProvideDifferingUuidOnSubsequentCall`.

FlorianZipser commented:
Sorry for the reformatting. I reverted them and disabled my Eclipse settings. Further I renamed the test. Please let me know, if there are further issues.

orien commented:
Thanks @FlorianZipser. Re the indentation: could you please replace the tabs with two spaces?

FlorianZipser commented:
I replaced the tabs with blanks and formatted both classes UuidGenerator and UuidGeneratorTest with a code style for Eclipse I found at https://github.com/google/styleguide/blob/gh-pages/eclipse-java-google-style.xml. Is this the proper style?
